07_safety/juvenile-arrests/juvenile-arrests-county-all.Rmd

```diff
@@ -0,0 +1,971 @@
+
+To test this second hypothesis, I repeat the join for 2022 only. There is a proportional number of unmatched observations from each data set for 2022 as compared to 2021-2022. So, I think this is okay.
+    todo(): Reviewer, does this logic make sense to you? I don't know what else I can do here to explain the unmatched observations. Many agencies did not report NIBRS data prior to 2021, so we expect a lot of missings.
+```{r}
+test_2022 <- left_join(
+  x = agency_weights %>% filter(year==2022),
+  y = arrests_agency %>% filter(year==2022), 
+  by = c("ori", "year")
+)
+
+```
+```{r}
+# Remove obsolete file
+rm(test_2022)
+```
+
+
+### 4.3 Summarize agency-level arrests to the county level
+Now that we have joined our agency-level juvenile arrests data onto the full universe of county-agencies, which links agencies to counties and assigns weights for agencies that serve more than one county, we can aggregate the counts from the agency to the county level.
+```{r}
+# Count number of arrests from reporting agencies in each county
+arrests_county <- arrests_agency_county %>%
+  group_by(year, GEOID, state, county) %>%
+  summarize(across(arr_total:age_15_17, ~sum(.x, na.rm=TRUE))) %>%
+  ungroup()
+
+```
+
+ 
+## 5. Calculate reporting rate for all agencies
+The quality indicator for this metric is based on the number of agencies reporting. We can't use the agency-level juvenile arrests data for this, because agencies may have zero juvenile arrests and thus be excluded but still have reported arrests for other ages. Instead, we use the full arrests file (i.e., all arrests and not just juvenile arrests) to calculate agency reporting rates.
+```{r}
+# Read appended juvenile arrests data back in for ease of future use
+# arrests_all <- read.csv(here::here("07_safety", "juvenile-arrests", "data", "arrests_all_21_22.csv"))
+
+# Sum total arrests by year and reporting agency
+arrests_all_agency <- arrests_all %>%
+  group_by(year, ori) %>%
+  summarize(arr_total_any = n()) %>%
+  ungroup()
+
+```
+
+Join total number of arrests for each agency onto the county/agency weight file.
+```{r}
+# Join by year and reporting agency
+arrests_all_agency_county <- left_join(
+  x = agency_weights,
+  y = arrests_all_agency,
+  by = c("year", "ori")
+)
+
+```
+
+There are a lot of unmatched observations. 17,397 observations are only in our universe of agencies with county-agency weights. My first thought is that since this is the universe of *all* agencies, some agencies may not have reported arrests and are therefore not in the arrests data. 1,985 observations are only in the 2021-2022 agency-level arrests data. My first thought is that since we only used the Batch Header File from 2022 to construct the universe of agencies and agency-county weights, it may be missing agencies from other years. 
```

Comment:
I re-ran the `01_agency_geo_county.qmd` file to make sure that the most recent versions of the files created in that program are available. I think what might have happened is that, because `01_agency_geo_county.qmd` and `01_agency_geo_place.qmd` are identical for both the reported-crime and juvenile-arrests metrics, I would just copy them from one branch to the other whenever I made changes to those files. But, for the files they create (e.g., `agency_weights` in this section) to update on the other branch, they also need to be ran on that branch. I think at one point I just copied the `01_agency_geo` programs without remembering to run them, so this branch had older versions of the files created by those programs.

Reply:
Amy, try running `01_agency_geo_county.qmd` again on this branch and see if the numbers still aren't matching. That program creates the file joined in this step, and I think you might have been working with an older version.